.github/scripts/test.sh

```diff
@@ -39,6 +39,10 @@ test_all_hls() {
         if ! [[ "${bin_noexe}" =~ "haskell-language-server-wrapper" ]] && ! [[ "${bin_noexe}" =~ "~" ]] ; then
             if ghcup install ghc --set "${bin_noexe/haskell-language-server-/}" ; then
                 "${hls}" typecheck "${test_module}" || fail "failed to typecheck with HLS for GHC ${bin_noexe/haskell-language-server-/}"
+
+                # After running the test, free up disk space by deleting the unneeded GHC version.
+                # Helps us staying beneath the 14GB SSD disk limit.
+                ghcup rm ghc "${bin_noexe/haskell-language-server-/}"
```

Comment:
:+1:

Reply:
@fendor this breaks the tests.

on line 51 we have 

```bash
    "$bindir/haskell-language-server-wrapper${ext}" typecheck "${test_module}" || fail "failed to typecheck with HLS wrapper"
```

Here are the logs:

```
2024-03-01T10:21:59.7648620Z + for hls in '"${bindir}/"haskell-language-server-*'
2024-03-01T10:21:59.7648970Z + bin=haskell-language-server-9.8.2~2.7.0.0
2024-03-01T10:21:59.7649280Z + bin_noexe=haskell-language-server-9.8.2~2.7.0.0
2024-03-01T10:21:59.7649730Z + [[ haskell-language-server-9.8.2~2.7.0.0 =~ haskell-language-server-wrapper ]]
2024-03-01T10:21:59.7650170Z + [[ haskell-language-server-9.8.2~2.7.0.0 =~ ~ ]]
2024-03-01T10:21:59.7650530Z + for hls in '"${bindir}/"haskell-language-server-*'
2024-03-01T10:21:59.7650840Z + bin=haskell-language-server-wrapper
2024-03-01T10:21:59.7651490Z + bin_noexe=haskell-language-server-wrapper
2024-03-01T10:21:59.7651910Z + [[ haskell-language-server-wrapper =~ haskell-language-server-wrapper ]]
2024-03-01T10:21:59.7652340Z + for hls in '"${bindir}/"haskell-language-server-*'
2024-03-01T10:21:59.7652680Z + bin=haskell-language-server-wrapper-2.7.0.0
2024-03-01T10:21:59.7653020Z + bin_noexe=haskell-language-server-wrapper-2.7.0.0
2024-03-01T10:21:59.7653490Z + [[ haskell-language-server-wrapper-2.7.0.0 =~ haskell-language-server-wrapper ]]
2024-03-01T10:21:59.7654410Z + /Users/github1/github/actions-runner/_work/haskell-language-server/haskell-language-server/.ghcup/bin/haskell-language-server-wrapper typecheck Data/ByteString.hs
2024-03-01T10:22:00.9380400Z Found "/private/var/folders/wb/zzxx2gnd7hb913crzw4m3h3w0000gp/T/hls_ci.XXXXXXX.YVU2SqT0L8/bytestring-0.11.1.0/hie.yaml" for "/private/var/folders/wb/zzxx2gnd7hb913crzw4m3h3w0000gp/T/hls_ci.XXXXXXX.YVU2SqT0L8/bytestring-0.11.1.0/a"
2024-03-01T10:22:00.9392480Z Run entered for haskell-language-server-wrapper(haskell-language-server-wrapper) Version 2.7.0.0 aarch64 ghc-9.2.8
2024-03-01T10:22:00.9395170Z Current directory: /private/var/folders/wb/zzxx2gnd7hb913crzw4m3h3w0000gp/T/hls_ci.XXXXXXX.YVU2SqT0L8/bytestring-0.11.1.0
2024-03-01T10:22:00.9396070Z Operating system: darwin
2024-03-01T10:22:00.9396430Z Arguments: ["typecheck","Data/ByteString.hs"]
2024-03-01T10:22:00.9399390Z Cradle directory: /private/var/folders/wb/zzxx2gnd7hb913crzw4m3h3w0000gp/T/hls_ci.XXXXXXX.YVU2SqT0L8/bytestring-0.11.1.0
2024-03-01T10:22:00.9400170Z Cradle type: Cabal
2024-03-01T10:22:00.9635970Z 
2024-03-01T10:22:00.9636830Z Tool versions found on the $PATH
2024-03-01T10:22:00.9639480Z cabal:          3.10.2.1
2024-03-01T10:22:00.9640080Z stack:          Not found
2024-03-01T10:22:00.9640450Z ghc:            Not found
2024-03-01T10:22:00.9640690Z 
2024-03-01T10:22:00.9640700Z 
2024-03-01T10:22:00.9641360Z Consulting the cradle to get project GHC version...
2024-03-01T10:22:00.9893730Z 2024-03-01T10:22:00.988616Z | Debug | cabal exec -v0 -- ghc --print-libdir
2024-03-01T10:22:00.9894360Z Failed to find the GHC version of this Cabal project.
2024-03-01T10:22:00.9895220Z Error when calling cabal exec -v0 -- ghc --print-libdir
2024-03-01T10:22:00.9895450Z 
2024-03-01T10:22:00.9896360Z Error: cabal: The program 'ghc' version >=7.0.1 is required but it could not
2024-03-01T10:22:00.9896730Z be found.
2024-03-01T10:22:00.9896830Z 
2024-03-01T10:22:00.9896830Z 
2024-03-01T10:22:00.9900060Z  2024-03-01T10:22:00.989765Z | Info | Starting server
2024-03-01T10:22:00.9901400Z 2024-03-01T10:22:00.990019Z | Error | Got EOF
```

If you analyze the logs, you see two things:

* there's no GHC to invoke (because you removed all of them)
* HLS wrapper doesn't seem to exit correctly when the toolchain is missing

This also fails more rigorously on runners that have GHC pre-installed.

Reply:
https://github.com/stable-haskell/haskell-language-server/actions/runs/8109118816/job/22164854919#step:6:11941

Reply:
> should we install some GHC version?

Up to you. In my CI, I install the recommended version.